### PR TITLE
SALTO-5455: support deploy of locale in zendesk

### PR DESCRIPTION
**Pull request by edenhassid**
support deploy of locale in zendesk

---

_Additional context for reviewer_
* hide the local ids in the account settings instance
* unhide the id in the locale instance as they are global and multienv friendly 

---
_Release Notes_: 
zendesk:
* add support for deploy of locales
---
_User Notifications_: 
_Replace me with a short sentence that describes changes that will appear in NaCls and are not caused by user actions (e.g. a new annotation, field values that are converted to references, etc). Hidden changes should not be listed._

### CODE REVIEW

File: packages/zendesk-adapter/src/change_validators/locale.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  changes
+    .filter(isModificationChange)
+    .map(getChangeData)
+    .filter(isInstanceElement)
+    .filter(instance => instance.elemID.typeName === LOCALE_TYPE_NAME)
+    .map(instance => ({
+      elemID: instance.elemID,
+      severity: 'Error',
+      message: 'Modification of locale is not supported',
+      detailedMessage: `Failed to update ${instance.elemID.getFullName()} since modification of locale is not supported`,
```
Comment (adfineberg) [inline packages/zendesk-adapter/src/change_validators/locale.ts:29]: Maybe add that this is a Zendesk limitation?

File: packages/zendesk-adapter/src/filters/locale.ts [hunk 2]
```diff
@@ -49,5 +64,85 @@ const filterCreator: FilterCreator = () => ({
...
+      return {
+        leftoverChanges,
+        deployResult: {
+          errors: [],
+          appliedChanges: [],
+        },
+      }
+    }
+
+    const [removalAndAdditionsChanges, modificationChanges] = _.partition(changes, isAdditionOrRemovalChange)
```
Comment (adfineberg) [inline packages/zendesk-adapter/src/filters/locale.ts:82]: I think you want to partition the `localeChanges` here, no?

**Review by adfineberg — COMMENTED** [review]
Looks great! 🕯️🔥 Added a couple of comments

File: packages/zendesk-adapter/src/filters/locale.ts (hunk 2 above)
Comment (edenhassid) [reply packages/zendesk-adapter/src/filters/locale.ts:82]: good catch!

**Review by adfineberg — APPROVED** [review]
🏆

**Pull request merged**